### PR TITLE
[API] Support list project background tasks

### PR DESCRIPTION
**Pull request by alonmr**
Add list project background tasks endpoint.
Support for filters (default filter is created in the last 7 days)
And check timeouts of background tasks

https://jira.iguazeng.com/browse/ML-5515

### CODE REVIEW

File: tests/api/api/test_background_tasks.py [hunk 1]
```diff
@@ -429,6 +429,47 @@ def test_trigger_migrations_from_worker_returns_same_response_as_chief(
...
+            == mlrun.common.schemas.BackgroundTaskState.succeeded
+        )
+        assert background_task.metadata.updated is not None
+
+    assert call_counter == 3
+
+
+# TODO: test list with timeout
+# TODO: test list stale background tasks (should be deleted)
+# TODO: test pagination?
```
Comment (liranbg) [inline tests/api/api/test_background_tasks.py:470]: not needed

File: tests/api/api/test_background_tasks.py [hunk 2]
```diff
@@ -429,6 +429,47 @@ def test_trigger_migrations_from_worker_returns_same_response_as_chief(
...
+        assert (
+            background_task.status.state
+            == mlrun.common.schemas.BackgroundTaskState.succeeded
+        )
+        assert background_task.metadata.updated is not None
+
+    assert call_counter == 3
+
+
+# TODO: test list with timeout
```
Comment (liranbg) [inline tests/api/api/test_background_tasks.py:468]: timeout of what?

**Review by liranbg — APPROVED** [review]

File: tests/api/api/test_background_tasks.py (hunk 1 above)
Comment (alonmr) [reply tests/api/api/test_background_tasks.py]: add filters instead

File: tests/api/api/test_background_tasks.py (hunk 2 above)
Comment (alonmr) [reply tests/api/api/test_background_tasks.py]: that's a note to self 😅 the timeout of the bacground task. added a test

**Pull request merged**